lib/velum/instance_type.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+require "json"
+
+module Velum
+  # Describe a public cloud instance type
+  class InstanceType
+    attr_reader :sort_value, :key, :category, :vcpu_count, :ram_bytes,
+      :ram_si_units, :details
+
+    def initialize(key, raw_data)
+      @key = key
+      @sort_value = raw_data["sort_value"]
+      @category = Velum::InstanceCategory.new(raw_data["category"])
+      @vcpu_count = raw_data["vcpu_count"]
+      @ram_bytes = raw_data["ram_bytes"]
+      @ram_si_units = raw_data["ram_si_units"]
+      @details = raw_data["details"]
+    end
+
+    def <=>(other)
+      sort_value <=> other.sort_value
+    end
+
+    def self.for(cloud)
+      raw_collection = get_raw_collection(cloud)
+      instance_type_collection_factory(
+        raw_collection["instance_types"],
+        raw_collection["categories"]
+      )
+    end
+
+    def self.get_raw_collection(cloud)
```

Comment:
i would not expose any other method than `self.for` through the interface. can we make them private?